iree/turbine/kernel/wave/scheduling/resources.py

```diff
@@ -47,17 +63,21 @@ class Operation(Enum):
     Operation.WRITE_GLOBAL: WRITE_GLOBAL_DELAY,
     Operation.MMA: MMA_DELAY,
     Operation.NOOP: 0,
+    Operation.VALU: VALU_DELAY,
+    Operation.SHUFFLE: SHUFFLE_DELAY,
 }
 
 # This table contains the resource usage for each operation.
 # Operations can use more than one resource for more than one cycle.
 resource_reservation_table = {
-    Operation.READ_SHARED: np.array([[0, 1, 0]]),
-    Operation.WRITE_SHARED: np.array([[0, 1, 0]]),
-    Operation.READ_GLOBAL: np.array([[1, 0, 0]]),
-    Operation.WRITE_GLOBAL: np.array([[1, 0, 0]]),
-    Operation.MMA: np.array([[0, 0, 1]]),
-    Operation.NOOP: np.array([[0, 0, 0]]),
+    Operation.READ_SHARED: np.array([[0, 1, 0, 0, 0]]),
```

Comment:
Not related to this PR, but does this mean we cannot read and write at the same time?

Reply:
This just specifies how many resources the instruction uses in one cycle. We can have multiple reads/writes by increasing the total number of resources available.